intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt

```diff
@@ -0,0 +1,51 @@
+package com.apollographql.ijplugin.util
+
+import com.apollographql.ijplugin.ApolloBundle
+import com.intellij.codeInsight.daemon.impl.analysis.DaemonTooltipsUtil
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptorBase
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+
+context(LocalInspectionTool)
+fun ProblemsHolder.registerProblem(
```

Comment:
🤯 Do we have context receivers in the codebase now?

Reply:
Ahaha yeah actually I've used them before in the plugin code 😅

Reply:
Plugin is probably OK but I'm not sure about the compatibility implication of those so I'd rather not use them in the libs

Reply:
Agreed 👍 We have a bit more leeway in the plugin.